dotnet-testing/UseMoqToReturnPassedInValue/UseMoqToReturnPassedInValue/PrinterQueue.cs

```diff
@@ -0,0 +1,37 @@
+﻿namespace UseMoqToReturnPassedInValue;
+
+public class PrinterQueue
+{
+    private readonly IPrinter _printer;
+
+    public PrinterQueue(IPrinter printer)
+    {
+        _printer = printer;
+    }
+
+    public int PrintAll(string[] jobs)
+    {
+        var characterCount = 0;
+
+        foreach (var job in jobs)
+        {
+            var result = _printer.Print(job);
+            characterCount += result.Length;
+        }
+
+        return characterCount;
+    }
+
+    public int PrintAllWithBug(string[] jobs)
+    {
+        var characterCount = 0;
+
+        foreach (var job in jobs)
+        {
+            var result = _printer.Print(jobs[0]); //deliberate bug
+            characterCount += result.Length;
+        }
+
+        return characterCount;
```

Comment:
This method isn't being called, and so really could be removed. Personally, since it is being listed as a "Why" use case in the article, doesn't mean that it needs to be in the repo.
Second, to make the "bug" a bit more realistic, I would change the foreach into a for loop, as in general we use an indexer within a foreach loop.

Reply:
I totally agree with what @jshergal has pointed out. Not all code should be in the repository, especially a code to show a deliberate bug. You can remove this from the repo but keep it in the article.